### ISSUE TITLE
Foundation watch throws error: ReferenceError: primordials is not defined

### ISSUE THREAD

**Issue by kevinmu17** [foundation/foundation-sites#12181]
## What should happen?
Foundation watch should start my dev project

## What happens instead?
Getting error:
`fs.js:45
} = primordials;
    ^

ReferenceError: primordials is not defined
    at fs.js:45:5
    at req_ (/usr/local/lib/node_modules/foundation-cli/node_modules/natives/index.js:143:24)
    at Object.req [as require] (/usr/local/lib/node_modules/foundation-cli/node_modules/natives/index.js:55:10)
    at Object.<anonymous> (/usr/local/lib/node_modules/foundation-cli/node_modules/graceful-fs/fs.js:1:37)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)`
...

## Possible Solution
clueless

How to reproduce:
1. Don't know, I bought a new laptop and installed everything completely new

## Your Environment
- Foundation version(s) used: foundation -v throws same error
- Browser(s) name and version(s): Google Chrome 87
- Device, Operating System and version: MacOS Big Sur 11.0.1
- Link to your project: n/a

## Checklist
- [x] I have read and follow the CONTRIBUTING.md document.
- [x] There are no other issues similar to this one.
- [x] The issue title and template are correctly filled.

**Comment by dtlevin** [comment]
Follow these instructions to fix this. 
https://timonweb.com/javascript/how-to-fix-referenceerror-primordials-is-not-defined-error/

1. In the same directory where you have package.json create an npm-shrinkwrap.json file with the following contents:
`{
  "dependencies": {
    "graceful-fs": {
        "version": "4.2.2"
     }
  }
}`

2. Run npm install, and don't worry, it'll update npm-shrinkwrap.json with a bunch of content.

3. Run ' npm start'

**Comment by DanielRuf** [comment]
See also:

https://stackoverflow.com/questions/56245622/foundation-referenceerror-primordials-is-not-defined-when-starting-a-foundat/56253873#56253873

https://github.com/foundation/foundation-sites/issues?q=is%3Aissue+sort%3Aupdated-desc+primordials+is%3Aclosed

**Comment by kevinmu17** [comment]
Thanks @dtlevin. But I use foundation to use foundation, i have like +/- 60 projects running so to hack arround all of them feels to off :) I see the issue still lies in the fact that foundation is still a decade behind node. so NVM did the job for me for now.

Are there any plan on updating everything to work with the @latest versions of all dependencies? (and fix all the errors?!)
a clean install foundation starts with **1209** vulnerabilities.

**Comment by DanielRuf** [comment]
Planned for the future. Closing as your issue is resolved now.

**Comment by DanielRuf** [comment]
> I started with `yarn add foundation-sites` - possibly a factor?

This adds foundations-sites as dependency, but is not the way for developing with it. The `start` script is in one of the template projects:

See https://github.com/foundation/foundation-zurb-template/ and https://github.com/foundation/foundation-sites-template/

**Comment by aderowbotham** [comment]
I see. As per the [other comments](https://github.com/foundation/foundation-sites/issues/12201) I also tried to install the **foundation-cli** but this is failing due to node version being 12.

**Comment by adaminfinitum** [comment]
I got the same error and was worried about trying to debug all this so (I don't use `Node.js` all that often) I completely uninstalled (`rm -rf`) every file and directory related to `node`, `npm`, `nvm` and anything that might cause me problems. I installed a fresh copy of Node.js LTS (14.17.6), which includes NPM (6.14.15).

I then ran `npm install --global foundation-cli` (it was successful at installing `foundation-cli@2.2.6`). So then I ran `foundation new` and got:
```
fs.js:45
} = primordials;
    ^

ReferenceError: primordials is not defined
    at fs.js:45:5
    at req_ (/usr/local/lib/node_modules/foundation-cli/node_modules/natives/index.js:143:24)
    at Object.req [as require] (/usr/local/lib/node_modules/foundation-cli/node_modules/natives/index.js:55:10)
    at Object.<anonymous> (/usr/local/lib/node_modules/foundation-cli/node_modules/graceful-fs/fs.js:1:37)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Module.require (internal/modules/cjs/loader.js:961:19)
    at require (internal/modules/cjs/helpers.js:92:18)
```

This is with everything being freshly installed, LTS versions, and following the tutorials exactly...with what should have installed Foundation 6.7.2. (which I resorted to out of sheer desperation because husky/commitlint won't let me commit changes to my clone of the repo from github...and I've had nothing but problems every time I have tried to use nested git repos [my project is using git(hub) and I don't want] a git repo of Foundation within a git repo for my project).

This is a current problem (as of Sept. 26, 2021) with everything being pristine, and cannot be due to Node v12, not working with Gulp v3, as many of the explanations online suggest as Node 12 isn't even installed on my machine.

**Comment by DanielRuf** [comment]
> This is a current problem (as of Sept. 26, 2021) with everything being pristine, and cannot be due to Node v12, not working with Gulp v3, as many of the explanations online suggest as Node 12 isn't even installed on my machine.

Your Node version is newer than the supported version. Please try / use Node 10. The issue affects Node 12 and newer in combination with gulp.

**Comment by DanielRuf** [comment]
You can use nvm to switch to Node 10.

**Comment by DanielRuf** [comment]
See also https://stackoverflow.com/a/56253873/753676

**Comment by adaminfinitum** [comment]
Please let it be that...I have been working on a new all-encompassing issue for the last hour, there is nothing I'd like more than to get this working an not waste anyone else's time...I'll try that and report back. Thank you so much!

**Comment by kevinmu17** [comment]
@adaminfinitum NVM (Node Version Manager) works [get it here](https://github.com/nvm-sh/nvm) and after installing it use the command `nvm use 10`. still it is a shame foundation is still not compatible with node 14. 

I think it's fair to say that I moved away from foundation due to lack of information about any future updates / maintenance of foundation (really sad about that but live goes on)

**Comment by adaminfinitum** [comment]
@DanielRuf, @kevinmu17  
Indeed, that was completely correct.

I uninstalled `Node.js` (and therefore `NPM`), and for that matter `NVM` earlier this year when the [dependency confusion vulnerability was disclosed](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610).

I typically only spend 10-20 percent of my professional time on development tasks, and with the pandemic, I hadn't used anything that relied on Node in probably 2 years.

Given the gravity of that vulnerability, I had no reason to update Node/NPM (and many reasons not to) and I just opted to delete it altogether.

Which is to say, I don't understand and appreciate the differences between Node 10 and 14...I think the last version I was aware of was 7 or 8 (LTS)...but 'stable' and 'LTS' typically indicate wide compatibility and acceptance.

As such, when I hit unexpected problems, I just deleted everything and reinstalled from scratch (which didn't help).

Using `NVM` to install `Node` v10 fixed every problem I had.

Thank you very much.
